`dns::record::ptr::by_ip`: Permit `zone` parameter to be empty

jearls opened the pull request:
... to allow fully-qualified `host` values.

Also create basic spec tests for dns`dns::record::ptr::by_ip` type

jearls commented:
This is an alternate proposal for #202

solarkennedy commented on manifests/record/ptr/by_ip.pp:
```diff
@@ -102,10 +104,16 @@
     zone => $reverse_zone,
     ttl  => $ttl ,
-    data => "${host}.${zone}"
+    data => $fqdn ,
```
Can you remove all these ` ,` (extra spaces before the commas)
I'm not sure there is a specific style guide rule to that, but it just looks kinda different to me

solarkennedy commented on manifests/record/ptr/by_ip.pp:
```diff
@@ -10,13 +10,16 @@
 #
 # * `$host`
-# The (unqualified) host pointed to by this ptr record.
+# The host pointed to by this ptr record.  This must be fully qualified
```
Can you clarify if there is supposed to be a period or not?

jearls commented:
Okay, I fixed the comments for both `$host` and `$zone` to indicate *no* trailing `.`.

I also cleaned up the spaces-before-commas.  I use that in my own modules because, to me, it's much more readable.  Most of the time, I remember not to do that when I'm working on other people's code.  Sorry about that.
